collection-loader.js

```diff
@@ -1,42 +1,95 @@
+const flatMap = require('lodash/flatMap');
+const {DEFAULT_STORY_FIELDS} = require('./constants');
 
 function loadCollectionItems(client, collections, {storyFields, storyLimits}) {
   const bulkRequestBody = collections.reduce(
-    (acc, collection) => Object.assign(acc, {
-      [collection.slug]: {
-        _type: "collection",
-        slug: collection.slug,
-        "story-fields": storyFields,
-        limit: storyLimits[get(collection, ["associated-metadata", "layout"])]
-      }
-    }), {});
+    (acc, collection) =>
+      Object.assign(acc, {
+        [collection.slug]: {
+          _type: 'collection',
+          slug: collection.slug,
+          'story-fields': storyFields,
+          limit:
+            storyLimits[get(collection, ['associated-metadata', 'layout'])],
+        },
+      }),
+    {}
+  );
 
-  return client.getInBulk({requests: bulkRequestBody}).then(response => response.results);
+  return client
+    .getInBulk({requests: bulkRequestBody})
+    .then(response => response.results);
 }
 
 // Ugly. This function updates all the items in place.
 // However, this is way more readable than a pure version
 function updateItemsInPlace(client, depth, items, {storyFields, storyLimits}) {
-  const collections = items.filter(item => item && item.type == "collection");
+  const collections = items.filter(item => item && item.type == 'collection');
 
-  if(depth == 0 || collections.length == 0)
-    return Promise.resolve();
+  if (depth == 0 || collections.length == 0) return Promise.resolve();
 
-  return loadCollectionItems(client, collections, {storyFields, storyLimits})
-    .then(collectionSlugToCollection => {
-      collections.forEach(collection => {
-        collection.summary = get(collectionSlugToCollection, [collection.slug, "summary"], '')
-        collection.items = get(collectionSlugToCollection, [collection.slug, "items"])
-      });
-      return updateItemsInPlace(client, depth - 1, flatMap(collections, collection => collection.items), {storyFields, storyLimits})
-    })
+  return loadCollectionItems(client, collections, {
+    storyFields,
+    storyLimits,
+  }).then(collectionSlugToCollection => {
+    collections.forEach(collection => {
+      collection.summary = get(
+        collectionSlugToCollection,
+        [collection.slug, 'summary'],
+        ''
+      );
+      collection.items = get(collectionSlugToCollection, [
+        collection.slug,
+        'items',
+      ]);
+    });
+    return updateItemsInPlace(
+      client,
+      depth - 1,
+      flatMap(collections, collection => collection.items),
+      {storyFields, storyLimits}
+    );
+  });
 }
 
-function loadNestedCollectionData(client, collection, {depth, storyFields, storyLimits}) {
-  return updateItemsInPlace(client, depth, collection.items, {storyFields, storyLimits})
-    .then(() => collection);
+function loadNestedCollectionData(
+  client,
+  collection,
+  {depth, storyFields, storyLimits, nestedCollectionStoryLimits}
```

Comment:
From where we are getting `nestedCollectionStoryLimits` ? 
i think you need to pass this var from `https://github.com/quintype/quintype-node-backend/blob/e036d7bd4ce8a0ea5897ea4b1afc5df186557ac9/index.js#L302`